### PR TITLE
Closes #10

### LINKED ISSUE #10: Add RethinkDB ORM

### PR DESCRIPTION
**Pull request by prank7**

### CODE REVIEW

File: server/config/database.js [hunk 1]
```diff
@@ -1,46 +1,20 @@
...
-    
-    cursor.toArray(function(err, result) {
-        if (err) throw err;
-        callback(null, result);
-    });
-	});
-}
+// Initiate the Database connection
+var thinky = require('thinky')(dbConfig);
+var r = require('rethinkdb'), assert = require('assert');
```
Comment (d4ncer) [inline server/config/database.js]: Try not to use comma separators for requiring vars into the local namespace. Declare them declaratively, i.e

``` javascript
var r = require('rethinkdb');
var assert = require('assert');
```

File: server/config/database.js [hunk 2]
```diff
@@ -1,46 +1,20 @@
...
 
-//Setup the Database and create tables, executed only when app starts.
-module.exports.setup = function() {
-  r.connect({host: dbConfig.host, port: dbConfig.port }, function (err, connection) {
+// Setup the Database
+thinky.databaseSetUp = function(){
+  r.connect({host: dbConfig.host, port: dbConfig.port}, function (err, connection) {
     assert.ok(err === null, err);
```
Comment (d4ncer) [inline server/config/database.js]: What exactly is assert doing here? Is it just checking if there is an error? If that's all it's doing, an if statement makes more sense than requiring an entire library just to check.

File: server/models/users.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+// Create the model
+var User = thinky.createModel("users", {
+  id: type.string(),
+  name: type.string(),
+  email: type.string()
+});
+
+// Define User API Endpoints
+var UserApi = {
+  getAll: function(req, res){
```
Comment (d4ncer) [inline server/models/users.js]: Why is a DB query sending back a response (and additionally, why does it have access to the req object)? You want to keep your DB/ORM calls separated from your router/controller logic. The router should ideally, on a request, call some DB function with some params (optionally, not required, depends on your query). The DB query should make the query, and return the data requested to the callback, which can then be processed by the controller/router.

File: server/api/users.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+var express = require('express');
+var router = express.Router();
+var User = require('../models/users');
+
+// GET /api/users/ 
+router.get('/', User.getAll);
```
Comment (d4ncer) [inline server/api/users.js]: The router logic here needs to be built out more, and separated from the DB/querying logic. See my comment on server/models/user.js

File: server/routes/about.js [hunk 5]
```diff
@@ -1,20 +1,9 @@
...
-			throw err;
-		else
-			res.render('about', {about: results});
-	};
-
-	//Pass a callback to getAbout function to fetch About data
-	database.getAbout(aboutData);
+// GET About page
+router.get('/', function(req, res){
+	res.render('about', {title: 'About'});
```
Comment (d4ncer) [inline server/routes/about.js:6]: Is there any reason to pass data to the app on render if you're using Flux/React on the client side for all rendering?

Comment (prank7) [inline server/routes/about.js:6]: React is being mounted inside the jade template but jade template itself expects 'title', thats why I am sending data. Isn't that the way to do it?

File: server/models/users.js (hunk 3 above)
Comment (prank7) [inline server/models/users.js]: Agree, Separation of concern. https://github.com/d4ncer/nugget/commit/715dc18baa50f7c471c614615198b39884aecb72 fixes that.

File: server/routes/about.js (hunk 5 above)
Comment (d4ncer) [inline server/routes/about.js:6]: Ah ok cool, that's fine.

**Pull request merged**